binary/reflect.go

```diff
@@ -803,6 +820,24 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, w io.Writer, n *int64,
 			WriteTo([]byte("}"), w, n, err)
 		}
 
+	case reflect.Map:
+		WriteTo([]byte("{"), w, n, err)
+		keys := rv.MapKeys()
```

Comment:
By c-style I just mean tendermint/binary's binary representation, as opposed to json-style.
Another option is to (re)add support for registering custom c-style/json-style write/read functions for types, and it's up to the implementor to ensure determinism and whether to support both c-style and json-style.  That seems pretty good, though it also seems a bit more challenging to implement.  Then we wouldn't need the map support, though we might want it in the future.